unity/Assets/Scripts/Unity/UnityTile.cs

```diff
@@ -25,13 +27,46 @@ void Start()
             tile = map.TileOnPoint(transform.position);
             
             SetComponents();
+            SetNature();
+            SetBehaviour();
 
             tile.TileChangedEvent += TileChanged;
         }
 
         public void TileChanged()
         {
             SetComponents();
+
+        }
+
+        private void SetBehaviour()
```

Comment:
This would've been perfect for a switch case, no? Can be refactored next sprint, no problem 👍